projects/components/src/splitter/splitter.service.ts

```diff
@@ -0,0 +1,106 @@
+    splitterElement: HTMLElement,
+    parentElement: HTMLElement,
+    direction: SplitterDirection
+  ): Observable<boolean> {
+    const moveOnParent$ = fromEvent<MouseEvent>(parentElement, 'mousemove');
+    const downOnHost$ = fromEvent<MouseEvent>(splitterElement, 'mousedown');
+    const upOnWindow$ = fromEvent<MouseEvent>(window, 'mouseup');
+
+    return downOnHost$.pipe(
+      mergeMap(() => moveOnParent$.pipe(takeUntil(upOnWindow$))),
+      map(event => {
+        const parentBounds = parentElement.getBoundingClientRect();
+        const prevSibling = this.getHTMLElement(splitterElement.previousElementSibling);
+        const nextSibling = this.getHTMLElement(splitterElement.nextElementSibling);
+
+        const previousElementUpdatedDimension = prevSibling
+          ? this.calculateUpdatedDimensionForPreviousElement(event, parentBounds, direction)
+          : 0;
+        const nextElementUpdatedDimension = nextSibling
+          ? this.calculateUpdatedDimensionForNextElement(previousElementUpdatedDimension, parentBounds, direction)
+          : 0;
+        const updatedDimensionData: UpdatedDimensionData[] = [];
+
+        if (prevSibling) {
+          updatedDimensionData.push({
+            element: prevSibling,
+            dimension: previousElementUpdatedDimension
+          });
+        }
+
+        if (nextSibling) {
+          updatedDimensionData.push({
+            element: nextSibling,
+            dimension: nextElementUpdatedDimension
+          });
+        }
+
+        return updatedDimensionData;
+      }),
+      map(data => this.maybeUpdateLayoutOfSiblings(data, direction))
+    );
+  }
+
+  private maybeUpdateLayoutOfSiblings(data: UpdatedDimensionData[], direction: SplitterDirection): boolean {
+    let layoutChanged: boolean = false;
+    data.forEach(datum => {
+      if (direction === SplitterDirection.Horizontal) {
+        this.updateWidth(datum.element, datum.dimension);
+      } else {
+        this.updateHeight(datum.element, datum.dimension);
+      }
+      layoutChanged = true;
+    });
+
+    return layoutChanged;
+  }
+
+  private getHTMLElement(element: Element | null): HTMLElement | undefined {
+    return element instanceof HTMLElement ? (element as HTMLElement) : undefined;
+  }
+
+  private calculateUpdatedDimensionForPreviousElement(
+    event: MouseEvent,
+    parentBounds: DOMRect,
+    direction: SplitterDirection
+  ): number {
+    return direction === SplitterDirection.Horizontal
+      ? Math.min(event.clientX - parentBounds.x, parentBounds.width)
+      : Math.min(event.clientY - parentBounds.y, parentBounds.height);
+  }
+
+  private calculateUpdatedDimensionForNextElement(
+    previousElementUpdatedDimension: number,
+    parentBounds: DOMRect,
+    direction: SplitterDirection
+  ): number {
+    return (
+      (direction === SplitterDirection.Horizontal ? parentBounds.width : parentBounds.height) -
+      previousElementUpdatedDimension -
+      4
```

Comment:
This is the splitter dom's width/height based on direction. This should be an input param .